packages/stateful/actions/components/AuthzAuthorization.tsx

```diff
@@ -0,0 +1,124 @@
+import { useFormContext } from 'react-hook-form'
+import { useTranslation } from 'react-i18next'
+
+import {
+  FormSwitchCard,
+  InputErrorMessage,
+  InputLabel,
+  KeyEmoji,
+  SelectInput,
+  TextInput,
+} from '@dao-dao/stateless'
+import { ActionComponent } from '@dao-dao/types/actions'
+import {
+  validateAddress,
+  validateContractAddress,
+  validateRequired,
+} from '@dao-dao/utils'
+
+import { AuthzExecActionTypes } from '../actions/AuthzExec'
+import { ActionCard } from './ActionCard'
+
+export interface AuthzOptions {}
+
+export const AuthzAuthorizationComponent: ActionComponent<AuthzOptions> = (
+  props
+) => {
+  const { t } = useTranslation()
+  const { data, fieldNamePrefix, onRemove, errors, isCreating } = props
+  const { register, setValue, watch } = useFormContext()
+
+  return (
+    <ActionCard
+      Icon={KeyEmoji}
+      onRemove={onRemove}
+      title={t('title.authzAuthorization')}
+    >
+      <div className="flex flex-col items-stretch gap-1">
+        <InputLabel name={t('form.grantOrRevokeAuthz')} />
+        <SelectInput
+          disabled={!isCreating}
+          fieldName={fieldNamePrefix + 'type_url'}
+          register={register}
+        >
+          <option value="/cosmos.authz.v1beta1.MsgGrant">
+            {t('form.grantAuthorizationOption')}
+          </option>
+          <option value="/cosmos.authz.v1beta1.MsgRevoke">
+            {t('form.revokeAuthorizationOption')}
+          </option>
+        </SelectInput>
+      </div>
+
+      <div className="flex flex-col items-stretch gap-1">
+        <InputLabel
+          name={t('form.granteeAddress')}
+          tooltip={t('form.granteeAddressTooltip')}
+        />
+        <TextInput
```

Comment:
`AddressInput` is better suited for wallet addresses, and if you pass in the stateful `ProfileDisplay` component to the `ProfileDisplay` prop of `AddressInput`, it'll display their profile!